### PR TITLE
[SWARM-1984] Refactoring Keycloak demo to show command and browser clients accessing Secured resource

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

File: security/keycloak/README.md [hunk 1]
```diff
@@ -51,38 +50,79 @@ docker run -it -d \
...
 
-## Obtain Token from Keycloak Server
+### Obtain Token from Keycloak Server
 
 ``` sh
 USER=user1
 PASS=password1
-RESULT=`curl -s --data "grant_type=password&client_id=curl&username=${USER}&password=${PASS}" http://localhost:8180/auth/realms/wildfly-swarm-keycloak-example/protocol/openid-connect/token`
+RESULT=`curl -s --data "grant_type=password&client_id=wildfly-swarm-cmd-client-example&username=${USER}&password=${PASS}" http://localhost:8180/auth/realms/wildfly-swarm-cmd-client/protocol/openid-connect/token`
 TOKEN=`echo $RESULT | sed 's/.*access_token":"//g' | sed 's/".*//g'`
```
Comment (Ladicek) [inline security/keycloak/README.md]: I think `jq` is much better.

File: security/keycloak/README.md [hunk 2]
```diff
@@ -51,38 +50,79 @@ docker run -it -d \
...
+
+``` sh
+THIS_EXAMPLE=/path/to/this/example
+java -Dswarm.keycloak.json.path=$THIS_EXAMPLE/src/main/resources/browser-client-keycloak.json \
+  -jar target/example-keycloak-swarm.jar
+```
+
+### Access the public client page
+
+Navigate the browser to "http:localhost:8080/"
```
Comment (Ladicek) [inline security/keycloak/README.md]: `http://`

File: security/keycloak/README.md [hunk 3]
```diff
@@ -51,38 +50,79 @@ docker run -it -d \
...
+Navigate the browser to "http:localhost:8080/" 
+
+### Access the secured resource
+
+Click on "Access Secured Resource"
+You'll get `Unauthorized` displayed.
+
+### Login
+
+Click on "Login" and use the "user1:password1" credentials to login into Keycloak.
```
Comment (Ladicek) [inline security/keycloak/README.md]: I'm not sure

> "user1:password1"

is a common format. Might be slightly misleading. I'd use

> `user1` / `password1`

Your call :-)

**Comment by sberyozkin** [comment]
I've removed webapp/META-INF/keycloak.json (it is now in resources/cmd-client-keycloak.json) - it always takes precedence even if SecuredArchivePreparer sets a custom keycloak.json. Significantly simplified the realm files. 
The only real 'added' complexity is the bigger README but I've tried to keep things as simple as possible.
Overall it should encourage users try more options with Swarm and KC.
Minor problem is that an example server has to be restarted to support different realms - a bit more work will be needed to support the multi-tenancy dynamically, will be a separate issue

**Comment by Ladicek** [comment]
I only took a cursory look, but I like this!

**Comment by sberyozkin** [comment]
Hi Ladislav, thanks, I'll take of the doc updates; FYI, the current token extraction script is the original one. I did try 'jq' while testing the secured booster, I recall though one needs to install an extra package, so I guess I'll just leave the way demo does it at the moment so that copying the whole block (starting from setting USER and PASSWORD vars and finishing with the token extraction) and pasting it into the shell works  OOB, hope you are OK with it

**Comment by sberyozkin** [comment]
I thought just now I could return to a single realm (file) with multiple clients, instead of having a realm file per client, but this requires 'relaxing' the keycloak.json the server starts with and dropping a 'resource' which points to a specific client. This is not possible though, 
http://lists.jboss.org/pipermail/keycloak-user/2017-August/011488.html

So I'll leave this PR as is, I guess showing multiple realms / tenancy would be of more interest to the users, as it seems to be asked about a lot by KC users.

**Pull request merged**